docs/book/products/catalog_promotions.rst

```diff
@@ -313,6 +313,24 @@ After changes in CatalogPromotion, we dispatch proper message with delay calcula
     To enable asynchronous Catalog Promotion, remember about running messenger consumer in a separate process, use the command: ``php bin/console messenger:consume main``
     For more information check official `Symfony docs <https://symfony.com/doc/current/messenger.html#consuming-messages-running-the-worker>`_
 
+Catalog promotion synchronicity
+-------------------------------
+
+TL;DR: There is no reason to use synchronous processing unless you have only a few products and manually activate and deactivate catalog promotions.
+
+Since the transport is asynchronous by default, in order to use synchronous processing, you need to update your ``.env`` configuration file:
+
+.. code-block:: bash
+
+    MESSENGER_TRANSPORT_DSN=sync://
+
```

Comment:
:thinking: I'm not sure we should recommend such change since it is a global config. Not just for this behaviour. This could clearly change some other processes.

If we want to propose a way to disable async for catalog promotion imo we should propose a way to do it just for this, not for the global app. Or am I mistaken ?

Reply:
I think you're right. I'm not well experienced with Symfony messenger but I did some digging and probably a better solution would be to reconfigure `messenger.yaml`.
```yaml
framework:
    messenger:
        # Uncomment this (and the failed transport below) to send failed messages to this transport for later handling.

        transports:
        # https://symfony.com/doc/current/messenger.html#transport-configuration
#            main: '%env(MESSENGER_TRANSPORT_DSN)%'
            main: 'sync://'
```
At this moment `main` transport is used only for catalog promotion related routings, therefore, it would not collide if we would introduce another async (sync) feature.

What do you think?